File: backend/project-service/src/main/java/com/connectexe/project/domain/entity/Project.java
package com.connectexe.project.domain.entity;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

import com.connectexe.project.domain.enums.DealType;
import com.connectexe.project.domain.enums.ProjectModerationStatus;
import com.connectexe.project.domain.enums.ProjectStage;
import com.connectexe.project.domain.enums.ProjectStatus;
import com.connectexe.project.domain.enums.ProjectVisibility;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "projects")
@Getter
@Setter
@NoArgsConstructor
public class Project {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @Column(name = "owner_id", nullable = false)
    private UUID ownerId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(length = 220, unique = true)
    private String slug;

    @Column(nullable = false)
    private String description;

    @Column
    private String summary;

    @Column
    private String content;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(nullable = false, columnDefinition = "project_stage")
    private ProjectStage stage;

    @Column(nullable = false, length = 120)
    private String industry;

    @Column(length = 2)
    private String country;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(nullable = false, columnDefinition = "project_status")
    private ProjectStatus status = ProjectStatus.DRAFT;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "moderation_status", nullable = false, columnDefinition = "project_moderation_status")
    private ProjectModerationStatus moderationStatus = ProjectModerationStatus.PENDING;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(nullable = false, columnDefinition = "project_visibility")
    private ProjectVisibility visibility = ProjectVisibility.PRIVATE;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(nullable = false, columnDefinition = "deal_type")
    private DealType dealType;

    @Column(name = "funding_target_usd", precision = 14, scale = 2)
    private BigDecimal fundingTargetUsd;

    @Column(name = "funding_need_usd", precision = 14, scale = 2)
    private BigDecimal fundingNeedUsd;

    @Column(name = "funding_raised_usd", precision = 14, scale = 2)
    private BigDecimal fundingRaisedUsd;

    @Column(name = "valuation_usd", precision = 14, scale = 2)
    private BigDecimal valuationUsd;

    @Column(name = "equity_percent", precision = 5, scale = 2)
    private BigDecimal equityPercent;

    @Column(name = "traction_summary")
    private String tractionSummary;

    @Column(name = "funding_timeline")
    private String fundingTimeline;

    @Column(name = "traction_metrics")
    private String tractionMetrics;

    @Column(name = "pitch_deck_url")
    private String pitchDeckUrl;

    @Column(name = "is_featured", nullable = false)
    private boolean featured = false;

    @Column(name = "featured_rank")
    private Integer featuredRank;

    @Column(name = "published_at")
    private OffsetDateTime publishedAt;

    @Column(name = "closed_at")
    private OffsetDateTime closedAt;

    @Column(name = "submitted_at")
    private OffsetDateTime submittedAt;

    @Column(name = "reviewed_by")
    private UUID reviewedBy;

    @Column(name = "reviewed_at")
    private OffsetDateTime reviewedAt;

    @Column(name = "created_at", nullable = false)
    @CreationTimestamp
    private OffsetDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    @UpdateTimestamp
    private OffsetDateTime updatedAt;

}

File: frontend/src/services/project.ts
import axios from 'axios'
import { tokenStorage } from './tokenStorage'

const projectApi = axios.create({
  baseURL: import.meta.env.VITE_PROJECT_API_URL ?? 'http://localhost:8082',
})

projectApi.interceptors.request.use((config) => {
  const token = tokenStorage.getAccessToken()
  if (token) {
    config.headers = config.headers ?? {}
    config.headers.Authorization = `Bearer ${token}`
  }
  return config
})

export interface Project {
  id: string
  ownerId: string
  title: string
  slug?: string | null
  description: string
  summary?: string | null
  content?: string | null
  stage: string
  industry: string
  country?: string | null
  status: string
  moderationStatus?: string | null
  visibility?: string | null
  dealType: string
  fundingTargetUsd?: number | null
  fundingNeedUsd?: number | null
  fundingRaisedUsd?: number | null
  valuationUsd?: number | null
  equityPercent?: number | null
  tractionSummary?: string | null
  fundingTimeline?: string | null
  tractionMetrics?: string | null
  pitchDeckUrl?: string | null
  featured: boolean
  featuredRank?: number | null
  tags?: string[]
  links?: ProjectLink[]
  media?: ProjectMedia[]
  publishedAt?: string | null
  closedAt?: string | null
  submittedAt?: string | null
  reviewedBy?: string | null
  reviewedAt?: string | null
}

export interface ProjectLink {
  id?: string
  type: string
  label?: string | null
  url: string
  sortOrder?: number | null
}

export interface ProjectMedia {
  id?: string
  fileUrl: string
  fileType?: string | null
  role?: string | null
  sortOrder?: number | null
  caption?: string | null
}

export async function listProjects(params: {
  stage?: string
  industry?: string
  country?: string
  dealType?: string
  status?: string
  moderationStatus?: string
  visibility?: string
}): Promise<Project[]> {
  const response = await projectApi.get('/api/projects', { params })
  return response.data.data
}

export async function getProject(id: string): Promise<Project> {
  const response = await projectApi.get(`/api/projects/${id}`)
  return response.data.data
}

export async function createProject(payload: {
  title: string
  description: string
  summary?: string
  content?: string
  stage: string
  industry: string
  dealType: string
  country?: string
  fundingTargetUsd?: number
  fundingNeedUsd?: number
  fundingRaisedUsd?: number
  valuationUsd?: number
  equityPercent?: number
  tractionSummary?: string
  fundingTimeline?: string
  tractionMetrics?: string
  pitchDeckUrl?: string
  tags?: string[]
  links?: ProjectLink[]
  media?: ProjectMedia[]
}): Promise<Project> {
  const response = await projectApi.post('/api/projects', payload)
  return response.data.data
}

export async function updateProject(id: string, payload: Partial<Project>): Promise<Project> {
  const response = await projectApi.put(`/api/projects/${id}`, payload)
  return response.data.data
}

export async function deleteProject(id: string): Promise<void> {
  await projectApi.delete(`/api/projects/${id}`)
}

export async function listMyProjects(): Promise<Project[]> {
  const response = await projectApi.get('/api/projects/mine')
  return response.data.data
}

export async function submitProject(id: string): Promise<Project> {
  const response = await projectApi.post(`/api/projects/${id}/submit`)
  return response.data.data
}

export async function approveProject(id: string): Promise<Project> {
  const response = await projectApi.put(`/api/projects/${id}/approve`)
  return response.data.data
}

export async function rejectProject(id: string): Promise<Project> {
  const response = await projectApi.put(`/api/projects/${id}/reject`)
  return response.data.data
}

export async function hideProject(id: string): Promise<Project> {
  const response = await projectApi.put(`/api/projects/${id}/hide`)
  return response.data.data
}

export async function listIndustries(): Promise<string[]> {
  const response = await projectApi.get('/api/projects/industries')
  return response.data.data
}
